.github/workflows/ci.yaml

```diff
@@ -10,7 +10,7 @@ on:
 
 jobs:
   main:
-    runs-on: ubuntu-latest
+    runs-on: ${{matrix.machine}}
```

Comment:
Sorry, I misled you. There is no need to change this.

Reply:
Dropped

Reply:
Just a note: the PR is in draft because I still have to fix the CI (as I wrote, I hope the signature error is only temporary).
I only pinged you for:
> any opinion about the Python version change?
I'm afraid it is a pretty big change, but maybe it was already discussed and 3.6 had been chosen for some reason I don't know